src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/MaterialForm/MaterialFormView.tsx

```diff
@@ -202,23 +204,26 @@ const MaterialFormView: React.FC<MaterialFormViewProps> = ({
                     variant="outlined"
                     error={!!errors.unitName}
                     helperText={errors.unitName?.message}
+                    value={field.value || ''}
+                    onChange={(event) => {
+                      const selectedValue = event.target.value;
+                      if (selectedValue === 'createUnit') {
+                        const unitName = prompt('Enter Unit Name');
+                        if (unitName) {
+                          createUnit(unitName);
+                        } else {
+                          event.target.value = field.value === undefined ? '' : field.value;
```

Comment:
this is making unitName: '', when you cancel, which causes an error to be thrown in the backend when you submit

Reply:
I believe this problem is fixed since I only call field.onChange when event.target.value !== ''. I had to do this conditional in line 215 since event.target.value cannot be set to undefined.